File: src/main/scala/nl/rug/jbi/search/ssap/model/Role.scala
package nl.rug.jbi.search.ssap.model

case class Role (val element:String, val name:String) {
  def toXml() = <role element={element} name={name}/>
}

object Role {
  def fromXml(node: scala.xml.Node):Role =
    new Role((node \ "@element").text, (node \ "@name").text)
}

File: src/main/java/nl/rug/jbi/search/ssap/model/Role.java
package nl.rug.jbi.search.ssap.model;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;

@XStreamAlias("role")
public class Role {

    @XStreamAsAttribute
    public final String name;
    @XStreamAsAttribute
    public final String element;

    public Role(String element, String name) {
        this.element = element;
        this.name = name;
    }
}
